### ISSUE TITLE
Wrong indentation: expected 6 but found 8 (indentation)

### ISSUE THREAD

**Issue by adoyle-h** [adrienverge/yamllint#35]
```yaml
---
- a:
    map:
        key: true
- b:
    map: false
```

I expect the snippet passed. But yamllint shows failure that "Wrong indentation: expected 6 but found 8 (indentation)" at line 4.

My yamllint configuration: 

```yaml
---
extends: default

rules:
    indentation:
        spaces: consistent
        indent-sequences: whatever
        check-multi-line-strings: true
```

Is there something wrong?

**Comment by adrienverge** [comment]
Hi @adoyle-h,

Yes, there is something wrong with your YAML snippet: the `map:` blocks aren't enough indented.

I tried to explain it in this diagram:
```
- a:
^ ^ map:
| | ^   key: true
| | |
| |  \
| |   start of the "map" block mapping, indented at 2
| |   from its parent block (the "a" block mapping)
| |   so effective indentation = 4 (instead of 2 + 4 = 6)
|  \
|   start of the "a" block mapping, with no
|   real indentation (it only follows the "- " notation)
|   but effective indentation = 2
 \
  start of the top-level block sequence
```

**Comment by adoyle-h** [comment]
Yes, I get it now. Thank you for your detailed explanation. @adrienverge
